lib/element/locator.js

```diff
@@ -52,6 +53,10 @@ class LocateElement {
 
     const elementInstance = Element.createFromSelector(selector, strategy);
 
+    if (isAppiumClient && strategy === 'id') {
+      return new By(strategy, selector);
+    }
```

Comment:
This change is for the following reason:

For `waitForElementVisible` command and `expect.element().visible` assertion, when we pass `{selector: 'com.app:id/web-login', locateStrategy: 'id'}` as selector, it gets modified to `{using: 'css selector', value: '*[id="com\\.app\\:id\\/web-login"]'` when searching for the element.

This fails in Appium v1.17.0 (default version of Appium in BrowserStack) since Appium v1.17.0 does not support `css selector` locate strategy, but works fine on the latest version of Appium 1.x.

So, I made this change to keep the locate strategy as `id` only when working with Appium.